src/Plugin/Field/FieldFormatter/StrawberryMediaFormatter.php

```diff
@@ -259,53 +258,57 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
+                // results site we will have for each Formatter one passed.
+                // Reason we use 'innode' array key using our $uniqueid
                 // @TODO probably better to use uuid() or the node id() instead of $uniqueid
-                $elements[$delta]['media'.$i]['#attributes']['data-iiif-infojson'] = $iiifpublicinfojson;
-                $elements[$delta]['media'.$i]['#attached']['drupalSettings']['format_strawberryfield']['openseadragon']['innode'][$uniqueid] = $nodeuuid;
-                $elements[$delta]['media'.$i]['#attached']['drupalSettings']['format_strawberryfield']['openseadragon'][$uniqueid]['width'] = $max_width_css;
-                $elements[$delta]['media'.$i]['#attached']['drupalSettings']['format_strawberryfield']['openseadragon'][$uniqueid]['dr:uuid']  = $file->uuid();
+                $elements[$delta]['media' . $i]['#attributes']['data-iiif-infojson'] = $iiifpublicinfojson;
+                $elements[$delta]['media' . $i]['#attached']['drupalSettings']['format_strawberryfield']['openseadragon']['innode'][$uniqueid] = $nodeuuid;
+                $elements[$delta]['media' . $i]['#attached']['drupalSettings']['format_strawberryfield']['openseadragon'][$uniqueid]['width'] = $max_width_css;
+                $elements[$delta]['media' . $i]['#attached']['drupalSettings']['format_strawberryfield']['openseadragon'][$uniqueid]['dr:uuid'] = $file->uuid();
                 // Used to keep annotations around during edit.
-                // Note: Since View modes are cached, if no change to the NODE this will be served from a cache! mmm.
+                // Note: Since View modes are cached, if no change to the NODE
+                // this will be served from a cache! mmm.
                 if ($this->currentUser->hasPermission('view strawberryfield webannotation') && $webannotations) {
                   $elements[$delta]['media' . $i]['#attached']['drupalSettings']['format_strawberryfield']['openseadragon'][$uniqueid]['keystoreid'] = WebAnnotationController::getTempStoreKeyName($fieldname, $delta, $nodeuuid);
                   $elements[$delta]['media' . $i]['#attached']['drupalSettings']['format_strawberryfield']['openseadragon'][$uniqueid]['webannotations'] = (boolean) $webannotations;
                   $elements[$delta]['media' . $i]['#attached']['drupalSettings']['format_strawberryfield']['openseadragon'][$uniqueid]['webannotations_tool'] = $webannotations_tool ? $webannotations_tool : 'rect';
-                  // This also never runs if cached. So after deletion we better call the controller!
+                  // This also never runs if cached. So after deletion we better
+                  // call the controller!
                   if (!empty($jsondata['ap:annotationCollection']) && is_array($jsondata['ap:annotationCollection'])) {
-                    $keystoreid = $elements[$delta]['media' . $i]['#attached']['drupalSettings']['format_strawberryfield']['openseadragon'][$uniqueid]['keystoreid'];
```

Comment:
This here seems to be a bug/of mine
WebAnnotationController::primeKeyStore() should be primed with the $keystoreid. Issue was this whole thing is cached and gets actually never call again after a cache clear. So wondering if it can be removed totally OR, better not and we need to prime it, at least once to avoid problems when the cache expires (e.g because of Node edit).

Can you please change to to WebAnnotationController::primeKeyStore($keystoreid) ?

Reply:
Done

Reply:
Almost! Sorry again (what a day!) its not $elements[$delta] its $items[$delta], $elements[$delta] its an render array, $items[$delta] the actual field with JSON data.

Reply:
Oops sorry about that!